src/services/grocerylist/GroceryListService.js

```diff
@@ -1,67 +1,273 @@
+            return await this.updateGroceryList(groceryList)
+
+        }
+        catch(err){
+            logError(err)
+            throw err
+        }
     }
 
     /**
-     * Remove an item from the grocery list if it exists
-     * @param {FoodItem} item 
+     * Remove an item at the index in the specified grocery list
+     * @param {GroceryList} groceryList the grocery list a food item is being removed from
+     * @param {Number} index the index at which the food item is located
+     * @returns {GroceryList} updated container if successful
+     * @throws error if the grocery list is not in the DB or he index is out of bounds 
      */
-    removeItem(item){
-        throw new Error('Not Implemented')
+    async removeFoodItemFromContainer(groceryList, index){
+        try{
+            this.__UseCollection(this.GROCERY_LIST_COLLECTION)
+
+            const groceryListToRemoveFrom = await this.getGroceryListById(groceryList.id)
+            if(groceryListToRemoveFrom == null)
+                throw new Error('the grocery list is not in the database')
+
+            if( index < 0 || groceryList.foodItems.length <= index)
+                throw new Error('the index is out of bounds')
+
+            groceryList.foodItems.splice(index, 1)
+            return await this.updateGroceryList(groceryList)
+        }
+        catch(err){
+            logError(err)
+            throw err
+        }
     }
 
     /**
-     * Gets an item from the grocery list if it exists
-     * @param {FoodItem} item 
-     * @returns a food item if found, otherwise null
+     * Gets all the food items from a grocery list
+     * @param {string} id the id of the grocery list to get the food items from
+     * @returns {FoodItem[]} an array of FoodItems that are currently in the grocery list
      */
-    getItem(item){
-        throw new Error('Not Implemented')
+    async getFoodItemArrayFromGroceryList(id){
+        try{
+            this.__UseCollection(this.GROCERY_LIST_COLLECTION)
+
+            const groceryList = await this.getGroceryListById(id)
+            if( groceryList == null)
+                throw new Error('the grocery list is not in the database')
+
+            const foodItemAry = [];
+
+            groceryList.foodItems.forEach(obj => {
+                const item = new FoodItem(obj.name, obj.photoURI, obj.quantity, obj.description, new Date(obj.expireDate), obj.nutritionData)
+                foodItemAry.push(item)
+            })            
+
+            return foodItemAry
+        }
+        catch(err){
+            logError(err)
+            throw err
+        }
     }
 
     /**
      * Finds and returns a list of food items who name
      *      matches the search
-     * @param {String} itemName
+     * @param {string} id the id of the grocery list being searched
+     * @param {string} itemName the name of the item being search for
      * @returns a list of food items 
      */
```

Comment:
I always shy away from using regex, but if it is thoroughly tested, I don’t see why not. Please make sure to add a comment as to what this regex is looking for. No one likes reading regex test <3